rust/src/checkpoints.rs

```diff
@@ -254,7 +254,7 @@ fn checkpoint_add_from_state(
 ) -> Result<Value, ArrowError> {
     let mut v = serde_json::to_value(action::Action::add(add.clone()))?;
 
-    if !add.partition_values.is_empty() && add.partition_values_parsed.is_none() {
```

Comment:
should we file an issue to track the work of supporting parsed partition values for better performance?

Reply:
Sure, I'll add an issue.